bndtools.m2e/src/bndtools/m2e/MavenRunProvider.java

```diff
@@ -32,14 +37,51 @@
 	private final MavenBndrunContainer mbc = new MavenBndrunContainer();
 
 	@Override
-	public Run create(IResource targetResource, RunMode mode) throws Exception {
+	public Bndrun create(IResource targetResource, RunMode mode) throws Exception {
 		if (!isMavenProject(targetResource)) {
 			return null;
 		}
 
+		final IMavenProjectFacade projectFacade = getMavenProjectFacade(targetResource);
+
+		Bndrun bndrun = create0(targetResource, projectFacade, mode);
+
+		Workspace workspace = bndrun.getWorkspace();
+
+		final MavenImplicitProjectRepository implicitRepo = new MavenImplicitProjectRepository( //
+			projectFacade, bndrun);
+
+		mavenProjectRegistry.addMavenProjectChangedListener( //
```

Comment:
Doesn't this duplicate work done in MavenImplicitProjectRunListener to add the listeners?

Reply:
yup, I goofed as I was reorganizing the changes.. the listener should only do the remove now.